native/cocos/platform/openharmony/modules/System.cpp

```diff
@@ -38,16 +39,64 @@ std::string System::getDeviceModel() const {
     return "";
 }
 
-ISystem::LanguageType System::getCurrentLanguage() const {
-    return LanguageType::ENGLISH;
+System::LanguageType System::getCurrentLanguage() const {
+    ccstd::string languageStr = getCurrentLanguageCode(); // NOLINT
+    if (languageStr == "en") {
+        return ISystem::LanguageType::ENGLISH;
+    } else if (languageStr == "zh") {
+        return ISystem::LanguageType::CHINESE;
+    } else if (languageStr == "fr") {
+        return ISystem::LanguageType::FRENCH;
+    } else if (languageStr == "it") {
+        return ISystem::LanguageType::ITALIAN;
+    } else if (languageStr == "de") {
+        return ISystem::LanguageType::GERMAN;
+    } else if (languageStr == "es") {
+        return ISystem::LanguageType::SPANISH;
+    } else if (languageStr == "du") {
+        return ISystem::LanguageType::DUTCH;
+    } else if (languageStr == "ru") {
+        return ISystem::LanguageType::RUSSIAN;
+    } else if (languageStr == "ko") {
+        return ISystem::LanguageType::KOREAN;
+    } else if (languageStr == "ja") {
+        return ISystem::LanguageType::JAPANESE;
+    } else if (languageStr == "hu") {
+        return ISystem::LanguageType::HUNGARIAN;
+    } else if (languageStr == "pt") {
+        return ISystem::LanguageType::PORTUGUESE;
+    } else if (languageStr == "ar") {
+        return ISystem::LanguageType::ARABIC;
+    } else if (languageStr == "no") {
+        return ISystem::LanguageType::NORWEGIAN;
+    } else if (languageStr == "pl") {
+        return ISystem::LanguageType::POLISH;
+    } else if (languageStr == "tr") {
+        return ISystem::LanguageType::TURKISH;
+    } else if (languageStr == "uk") {
+        return ISystem::LanguageType::UKRAINIAN;
+    } else if (languageStr == "ro") {
+        return ISystem::LanguageType::ROMANIAN;
+    } else if (languageStr == "bg") {
+        return ISystem::LanguageType::BULGARIAN;
+    }
+    return ISystem::LanguageType::ENGLISH;
```

Comment:
JS returns result with `return i18n.getSystemLanguage();`. Then why need to implement in C++?

Reply:
To keep the interface consistent. Both interfaces are exported for use by the cocos engine.
And the value returned by i18n.getSystemLanguage() is a little different than we expected.

Reply:
> And the value returned by i18n.getSystemLanguage() is a little different than we expected.

If so, why use it in js?

Reply:
i18n.getSystemLanguage() is called for C++, which converts the expected values and returns them to the ts engine.
In addition, i18n is an internal module and interface of openharmony.